libsolidity/interface/StandardCompiler.cpp

```diff
@@ -719,8 +723,11 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 			if (!smtlib2Responses.isObject())
 				return formatFatalError("JSONError", "\"auxiliaryInput.smtlib2responses\" must be an object.");
 
-			for (auto const& hashString: smtlib2Responses.getMemberNames())
+			for (auto const& key: smtlib2Responses.getMemberNames())
 			{
+				auto delimiterPos = key.find(queryKeyDelimiter);
+				auto hashStart = delimiterPos == string::npos ? 0 : delimiterPos + 1;
```

Comment:
Shouldn't we rather assert that the delimiter is there?

Reply:
That's possible, I thought it's better to be more benevolent. At this point the prefix is not needed. The only place where it is needed is to fix order in the JSON object.

Reply:
Ok, that's also fine